etc/generate-envoy-config/envoy.jsonnet

```diff
@@ -0,0 +1,196 @@
+// Services that we want to proxy; where they are and how to route them.
+local services = {
+  'pachd-grpc': {
+    internal_port: 1650,
+    external_port: 30650,
+    grpc: true,
+    service: 'pachd-proxy-backend',
+    routes: [
+      {
+        match: {
+          grpc: {},
+          prefix: '/',
+        },
+        route: {
+          cluster: 'pachd-grpc',
+          timeout: '604800s',
+        },
+      },
+    ],
+    health_check: {
+      grpc_health_check: {},
+      healthy_threshold: 1,
+      interval: '10s',
+      timeout: '10s',
+      unhealthy_threshold: 2,
+      no_traffic_interval: '10s',
+      no_traffic_healthy_interval: '10s',
+    },
+  },
+  'pachd-s3': {
+    internal_port: 1600,
+    external_port: 30600,
+    service: 'pachd-proxy-backend',
+    routes: local base = {
+      route: {
+        cluster: 'pachd-s3',
+        idle_timeout: '600s',
+        timeout: '604800s',
+      },
+    }; [
+      base {  // S3v4
+        match: {
+          prefix: '/',
+          headers: [
+            {
+              name: 'authorization',
+              string_match: {
+                prefix: 'AWS4-HMAC-SHA256',
```

Comment:
where is this prefix passed for S3?

Reply:
S3 V4 sends authorization headers like: 

```
Authorization: AWS4-HMAC-SHA256 Credential=V4Ptmxchcj04TY5vmngJAD0RiuJ3JYc6/20220517/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
```

S3 V2 sends authorization headers like:

```
Authorization: AWS **REDACTED**:**REDACTED**
```

Those are the only protocols that "S2" (our in-house S3 clone) supports.

Obviously this is somewhat flawed in that if you wanted to talk to console and stuck "Authorization: AWS is not in use" in the HTTP request, it wouldn't be routed correctly.